tracing/tracing/src/main/java/io/helidon/tracing/SpanLifeCycleListener.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.tracing;
+
+/**
+ * Behavior of listeners notified of span life cycle events.
+ * <p>
+ * The interface declares default empty implementations for each method, allowing concrete implementations of the
+ * interface to implement only the methods that are relevant for them.
+ * <p>
+ * Helidon invokes the applicable methods of a life cycle listener in the following order:
```

Comment:
If you decide to keep the notion of lifecycle: "life cycle" ▶️ "lifecycle"
If you decide to lose it: "a life cycle listener" ▶️ "an implementation of this interface"

Reply:
Changed